### PR TITLE
Bugfix/#2723 self label checkboxes

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**

UI/UX issue #2723 requests that all check boxes be made "self-labelling".  This makes the change to the relevant forms.

(It also causes the Server UI to start on "Server Setup" rather than "Options" consistently.)

CHANGELOG: Make checkboxes self-labelling

**Context: Fixes an issue?**

Fixes: #2723 

**Does this change need documentation? What needs to be documented and how?**

Yes, User Manual and Server Manual screen shot updates are needed.  This affects translations.

**Status of this Pull Request**

Needs review.
Depends on:
- bugfix/remove-remaining-directory-server-mentions
- bugfix/rpc-remaining-directory-server-mentions

**What is missing until this pull request can be merged?**

Needs review.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: src/clientsettingsdlgbase.ui [hunk 1]
```diff
@@ -1151,6 +1201,9 @@
                   <property name="alignment">
                    <set>Qt::AlignCenter</set>
                   </property>
+                  <property name="buddy">
```
Comment (ann0see) [inline src/clientsettingsdlgbase.ui:1204]: Does name=buddy mean? No translation required or what does it stand for?

**Review by ann0see — COMMENTED** [review]
Thanks. Will try if I have time.

File: src/clientsettingsdlgbase.ui (hunk 1 above)
Comment (pljones) [reply src/clientsettingsdlgbase.ui:1204]:
```
$ grep -n edtAlias src/clientsettingsdlgbase.ui
92:                     <cstring>pedtAlias</cstring>
141:                   <widget class="QLineEdit" name="pedtAlias">
1344:  <tabstop>pedtAlias</tabstop>
```
The "buddy" is the field this field labels.  It's the _internal_ field name - i.e. what's used in the C++ code, not what's displayed.  All the label widgets _should_ have one.

**Comment by pljones** [comment]
> Thanks. Will try if I have time.

Thanks.

**Comment by pljones** [comment]
The "My Profile" tab allows the user to adjust the width of the window.
image
On Windows (GitHub build), using the drop down doesn't set itself to the correct width -- and I've no idea why.
image

The window shrinks width-ways too much
image

When it's stretching width-ways, some of the labels become unaligned - I did have a version with this fixed but I lost it somehow.
Aligned:
image
Unaligned:
Windows: image
Linux seems somewhat more prone to it (probably a skin-dependent issue as to _how much_ it happens): image

**Comment by ann0see** [comment]
> The window shrinks width-ways too much

Not a regression I think. Someone requested it a while ago. But not sure why.

**Comment by ann0see** [comment]
grafik


I think that's ok?

**Comment by pljones** [comment]
> > The window shrinks width-ways too much
> 
> Not a regression I think. Someone requested it a while ago. But not sure why.

Yeah - the "form" version stopped it but broke the layout in other ways (i.e. stretch all fields equally).

File: src/serverdlg.cpp [hunk 2]
```diff
@@ -115,10 +118,9 @@ CServerDlg::CServerDlg ( CServer* pNServP, CServerSettings* pNSetP, const bool b
     cbxLocationCountry->setAccessibleName ( tr ( "Combo box for location of this server" ) );
 
     // enable recorder
-    chbEnableRecorder->setAccessibleName ( tr ( "Checkbox to turn on or off server recording" ) );
-    chbEnableRecorder->setWhatsThis ( "<b>" + tr ( "Enable Recorder" ) + ":</b> " +
-                                      tr ( "Checked when the recorder is enabled, otherwise unchecked. "
-                                           "The recorder will run when a session is in progress, if (set up correctly and) enabled." ) );
+    chbJamRecorder->setAccessibleName ( tr ( "Checkbox to turn on or off server recording" ) );
+    chbJamRecorder->setWhatsThis ( "<b>" + tr ( "Jam Recorder" ) + ":</b> " +
+                                   tr ( "The recorder will run when a session is in progress, if (set up correctly and) checked." ) );
```
Comment (mcfnord) [inline src/serverdlg.cpp:123]:
```suggestion
                                   tr ( "If checked, the recorder will run when a session is in progress, if recording is set up correctly." ) );
```

File: src/translation/translation_de_DE.ts [hunk 3]
```diff
@@ -1646,12 +1646,12 @@ Wir haben Deinen Kanal stummgeschaltet und die Funktion &apos;Stummschalten&apos
...
         <location filename="../clientsettingsdlg.cpp" line="394"/>
         <source>Audio Alerts</source>
-        <translation>Audiohinweise</translation>
+        <translation type="unfinished">Audiohinweise</translation>
     </message>
     <message>
         <location filename="../clientsettingsdlg.cpp" line="395"/>
-        <source>Enable audio alert when receiving a chat message and when a new client joins the session. A second sound device may be required to hear the alerts.</source>
-        <translation>Aktiviert Audiohinweise beim Empfang einer Chat-Nachricht bzw. wenn ein neuer Client beitritt. Möglicherweise ist ein zweites Audiogerät erforderlich, um die Hinweise zu hören.</translation>
+        <source>Trigger an audio alert when receiving a chat message and when a new client joins the session. A second sound device may be required to hear the alerts.</source>
```
Comment (mcfnord) [inline src/translation/translation_de_DE.ts:1653]:
```suggestion
        <source>Trigger an audio alert when receiving a chat message and when a new client joins the session. (A second sound device may be required to hear the alerts.)</source>
```

I'm going through a (parenthetical) phase.

File: src/translation/translation_de_DE.ts [hunk 4]
```diff
@@ -3854,13 +3854,13 @@ Wir haben Deinen Kanal stummgeschaltet und die Funktion &apos;Stummschalten&apos
...
-        <source>Enable Recorder</source>
-        <translation>Aktiviere die Aufnahmefunktion</translation>
+        <source>Jam Recorder</source>
+        <translation type="unfinished">Aktiviere die Aufnahmefunktion</translation>
     </message>
     <message>
         <location filename="../serverdlg.cpp" line="120"/>
-        <source>Checked when the recorder is enabled, otherwise unchecked. The recorder will run when a session is in progress, if (set up correctly and) enabled.</source>
-        <translation>Angehakt, wenn die Aufnahmefunktion aktiviert ist. Die Aufnahme wird automatisch gestartet, wenn eine Jam-Session läuft.</translation>
+        <source>The recorder will run when a session is in progress, if (set up correctly and) checked.</source>
```
Comment (mcfnord) [inline src/translation/translation_de_DE.ts:3862]:
```suggestion
        <source>If checked, the recorder will run when a session is in progress, if recording is set up correctly.</source>
```

File: src/translation/translation_fr_FR.ts [hunk 5]
```diff
@@ -1730,12 +1730,12 @@ Nous avons coupé votre canal et activé &quot;Me silencer&quot;. Veuillez d&apo
...
         <location filename="../clientsettingsdlg.cpp" line="394"/>
         <source>Audio Alerts</source>
-        <translation>Alertes audio</translation>
+        <translation type="unfinished">Alertes audio</translation>
     </message>
     <message>
         <location filename="../clientsettingsdlg.cpp" line="395"/>
-        <source>Enable audio alert when receiving a chat message and when a new client joins the session. A second sound device may be required to hear the alerts.</source>
-        <translation>Activez l&apos;alerte audio lors de la réception d&apos;un message de chat et lorsqu&apos;un nouveau client rejoint la session. Un deuxième périphérique sonore peut être nécessaire pour entendre les alertes.</translation>
+        <source>Trigger an audio alert when receiving a chat message and when a new client joins the session. A second sound device may be required to hear the alerts.</source>
```
Comment (mcfnord) [inline src/translation/translation_fr_FR.ts:1737]:
```suggestion
        <source>Trigger an audio alert when receiving a chat message and when a new client joins the session. (A second sound device may be required to hear the alerts.)</source>
```

File: src/translation/translation_de_DE.ts (hunk 3 above)
Comment (pljones) [reply src/translation/translation_de_DE.ts:1653]: Don't edit the `.ts` files, edit the source.

File: src/translation/translation_fr_FR.ts (hunk 5 above)
Comment (pljones) [reply src/translation/translation_fr_FR.ts:1737]: As above.

File: src/serverdlg.cpp (hunk 2 above)
Comment (pljones) [reply src/serverdlg.cpp:123]: Maybe this way around?
```suggestion
                                   tr ( "If checked and set up correctly, the recorder will run when a session is in progress." ) );
```

**Comment by pljones** [comment]
I've got a new version of https://github.com/pljones/jamulus/tree/bugfix/%232723-self-label-checkboxes-forms2 available here:
https://github.com/pljones/jamulus/actions/runs/4616999421

The Windows version still has this annoying display error:
image
The window needs stretching about six pixels wider to get the dropdown to display correctly.  I'd rather it "naturally" found the right width, as different font sizes (i.e. themes) may have different needs -- so I was wondering if this is a Qt 5 bug.

Can anyone - @jamulussoftware/maindevelopers particularly - take a look?

Also, @ann0see, @gilgongo or @emlynmac, could you check that build on a Mac, please.  If it's okay (this is the **Client Settings Dialog** and **Server Dialog** changes) - including when resizing, please confirm here.  I'll rebase and make it https://github.com/pljones/jamulus/tree/bugfix/%232723-self-label-checkboxes if it's okay, then we can progress this (once the [directory changes](https://github.com/jamulussoftware/jamulus/pull/3017) are merged).

**Comment by pljones** [comment]
In addition to the build here:
- `forms1` version: https://github.com/pljones/jamulus/actions/runs/4725240278
- `forms2` version: https://github.com/pljones/jamulus/actions/runs/4725241301

I could do with MacOS people checking both to see whether they work (profile, maximize window).  I'd prefer to go with `forms2`.

**Comment by gilgongo** [comment]
> In addition to the build here:
> 
> * `forms1` version: https://github.com/pljones/jamulus/actions/runs/4725240278
> * `forms2` version: https://github.com/pljones/jamulus/actions/runs/4725241301

Just tried forms2 on Monterey 12.6.1 and it seems OK to me (profile, maximize window)

**Comment by ann0see** [comment]
@gilgongo, could you please provide screenshots? On Linux everything looks fine.

**Comment by gilgongo** [comment]
Mind you, I don't know what it's intended look like exactly:

Screenshot 2023-04-19 at 10 07 02 pm

Screenshot 2023-04-19 at 10 08 41 pm

**Comment by ann0see** [comment]
I think that's not the intended layout. It should look like this: https://github.com/jamulussoftware/jamulus/pull/2996#issuecomment-1399320534

**Comment by gilgongo** [comment]
Ah OK. BTW "Mixer Rows" would be nicer at about 25% of the length it is in that screenshot, but it's not a big problem.

**Comment by pljones** [comment]
No, that's not the intended layout.  That's why it was important to test on macOS explicitly as Linux and Windows do not show that behaviour.

**Comment by pljones** [comment]
OK, reverted that.

**Comment by ann0see** [comment]
So @pljones do you think the macOS build is ok with your latest changes?

**Comment by pljones** [comment]
> So @pljones do you think the macOS build is ok with your latest changes?

I've not had time over the last few days to keep up.  _Hopefully_ the current build on this branch is okay again.

**Comment by ann0see** [comment]
I will - hopefully - have access to a real Mac in the next weeks but no guarantees.

**Comment by pljones** [comment]
OK, it builds and runs on Linux (6.5.1).  One tiny glitch (likely 6.5.1 related) - opening settings, the "My Profile" didn't show until I switched tabs.  Very odd...

**Comment by pljones** [comment]
> OK, it builds and runs on Linux (6.5.1). One tiny glitch (likely 6.5.1 related) - opening settings, the "My Profile" tab didn't show until I switched tabs. Very odd...

Confirmed on `main` - 
Check settings:
image
No "My Profile tab"
image
Switch to "Advanced Setup" and it appears:
image
Once it's appeared, it stays for the duration of the run.  Checking the checkbox multiple times doesn't fix it.  Using the "Settings" menu to go straight to "My Profile" or using Ctrl-P both work okay.

So that'll need picking up under the move to Qt6.5.1...

**Comment by ann0see** [comment]
Screenshot my profile

This is the current artifact from this PR - and I believe it's ok.

File: src/serverdlg.cpp [hunk 6]
```diff
@@ -37,6 +37,9 @@ CServerDlg::CServerDlg ( CServer* pNServP, CServerSettings* pNSetP, const bool b
 
     setupUi ( this );
 
+    // always start on the main tab
+    tabWidget->setCurrentIndex ( 0 );
+
     // set window title
```
Comment (ann0see) [inline src/serverdlg.cpp:43]: Does the Qt 6.5 bug also occur if this isn't set?

**Review by ann0see — APPROVED** [review]
Approving on the basis that this works on the CURRENT Qt version.

File: src/serverdlg.cpp (hunk 6 above)
Comment (pljones) [reply src/serverdlg.cpp:43]: That's `src/serverdlg.cpp`, not related to the client settings "My Profile" weirdness.

Comment (ann0see) [reply src/serverdlg.cpp:43]: Uh. Yes. It's a different file - so that's true.

**Comment by pljones** [comment]
Merging - we can pick up any reported problems later.  Hopefully it'll be okay.

**Comment by ann0see** [comment]
@pljones does this still need documentation? If no, please drop the needs documentation label.

**Pull request merged**